rad/src/error.rs

```diff
@@ -106,3 +129,46 @@ impl From<std::num::ParseFloatError> for RadError {
         }
     }
 }
+
+impl Into<jsonrpc_core::Error> for RadError {
+    fn into(self) -> jsonrpc_core::Error {
+        let build_error = |rad_error: RadError, code: i64| {
+            let mut err =
+                jsonrpc_core::types::error::Error::new(jsonrpc_core::ErrorCode::from(code));
+            err.message = rad_error.to_string();
+            err
+        };
+
+        match &self {
```

Comment:
I would move this match to a `RadError::code` method:

```rust
impl RadError {
    fn code(&self) -> i64 {

    }
}
```

And then here it would just be

```rust
build_error(self, self.code())
```